src/userscriptvalidators.ts

```diff
@@ -0,0 +1,71 @@
+            }
+            break;
+        case 'bool':
+            if (typeof setting.value !== "boolean") {
+                settingIsMalformed = `'${setting.value}' (${typeof setting.value}) is NOT a valid value for setting type '${setting.type}.'`
+            }
+            break;
+        case 'sel':
+            if (!('opts' in setting)) {
+                settingIsMalformed = `Setting type 'sel' requires property .opts as an array with each option for this setting. (Example: {opts: ['option1', 'option2'...]})`
+            } else if (!Array.isArray(setting.opts)) {
+                settingIsMalformed = `Setting type 'sel' requires the .opts property to be an ARRAY! (Example: {opts: ['option1', 'option2'...]})`
+            } else if (setting.opts.find(option => !['number', 'string'].includes(typeof option))) {
+                settingIsMalformed = `All options (.opts) in setting type 'sel' need to be either NUMBERs or STRINGs.`
+            } else if (setting.opts.find(option => typeof option !== typeof setting.opts[0])) {
+                settingIsMalformed = `All options (.opts) in setting type 'sel' need to be the same type! (You cannot have both STRING and NUMBER options.)`
+            } else if (setting.opts.length < 2) {
+                settingIsMalformed = `Setting type 'sel' must have at least 2 options to choose from!`
+            } else if (!setting.opts.includes(setting.value)) {
+                settingIsMalformed = `Setting type 'sel' must have its value set to one of the options defined in .opts!`
+            }
+            break;
+        case 'color':
+            if (typeof setting.value !== "string") {
+                settingIsMalformed = `'${setting.value}' (${typeof setting.value}) is NOT a valid value for setting type '${setting.type}.'`
+            } else {
+                if (!setting.value.match(/^#([0-9a-fA-F]{3}){2}$/g)) {
+                    settingIsMalformed = `'${setting.value}' is not a valid color. Use #ffffff`
+                }
+            }
+            break;
+        default:
+            settingIsMalformed = `'${setting.type}' is NOT a valid setting type.`
+            break;
+    }
+    if (('title' in setting) && setting.title.length < 1) {
+        settingIsMalformed = `'${setting?.title}' is NOT a valid setting title.`
+    }
+    if (!('changed' in setting)) {
+        settingIsMalformed = `Custom setting requires .changed() function property.`
+    } else if (typeof setting.changed !== 'function') {
+        settingIsMalformed = `Custom setting .changed property MUST be a function. (Example: {changed: (value) => {}})`
+    }
+
+    return settingIsMalformed
+}
+
+export const customSettingSavedJSONIsMalformed = function (settingKey: string, settings: Record<string, UserscriptRenderReadySetting>, settingsJSON: Record<string, UserscriptRenderReadySetting>): boolean {
```

Comment:
good good but the function name is a bit misleading: it implies it would return true if the setting JSON is malformed, but it returns true if the JSON is valid. 
please consider some different name,maybe like `isCustomSettingJSONValidFormat`